app/controllers/charges_controller.rb

```diff
@@ -9,25 +9,36 @@ def new
   end
 
   def create
+    # I know this controller action is fat. It will be refactored in a future PR
     order = current_user.orders.pending.last
     if order
       # Amount in cents
       amount = order.stripe_total
-      customer = Stripe::Customer.create(
-        email:  current_user.email,
-        source: params[:stripeToken]
-      )
+
+      if session[:stripeToken]
+        customer = Stripe::Customer.create(
+          email:  current_user.email,
+          source: session[:stripeToken]
+        )
+        session[:stripeToken] = nil
+        customer_id = customer.id
+      else
+        customer_id = current_user.stripe_token
+      end
+
+      current_user.save_stripe_token(customer_id)
```

Comment:
I don't think it is necessary to have a method for updating one attribute.
`current_user.update_attribute(:stripe_token, customer_id)`